TCAT/Controllers/RouteDetail+ContentViewController.swift

```diff
@@ -466,6 +515,76 @@ class RouteDetailContentViewController: UIViewController {
                 setIndex(of: marker, with: waypoint.wpType)
                 bounds = bounds.includingCoordinate(waypoint.coordinate)
             }
+            
+            if let busPath = path as? BusPath {
+                // Create circles at the first and last coordinate points / stops for bus route(s)
+                if let startBusStopCoordinate = busPath.waypoints.first {
+                    let startCircle = busStopCircles(at: startBusStopCoordinate.coordinate, on: mapView)
+                    finalDestinationCircles.append(startCircle)
+                }
+                
+                if let finalBusStopCoordinate = busPath.waypoints.last {
+                    let endCircle = busStopCircles(at: finalBusStopCoordinate.coordinate, on: mapView)
+                    finalDestinationCircles.append(endCircle)
+                }
+            }
+            
+            // Extract and append all coordinates of waypoints
+            if let walkPath = path as? WalkPath {
+                for circleInfo in walkPath.circles {
+                    let circle = GMSCircle(position: circleInfo.coordinate, radius: circleInfo.radius)
+                    if pathCount == 0 {
+                        firstRouteSegment.append(circle)
+                    } else {
+                        finalRouteSegment.append(circle)
+                    }
+                }
+            }
+            pathCount += 1
+        }
+        
+        func mapRouteSegment(_ segment: [GMSCircle], to path: GMSMutablePath, addMarker: Bool = false) {
+            for (index, waypoint) in segment.enumerated() {
+                let coordinates = CLLocation(latitude: waypoint.position.latitude, longitude: waypoint.position.longitude)
+                path.addLatitude(coordinates.coordinate.latitude, longitude: coordinates.coordinate.longitude)
+                if addMarker && index == segment.count - 1 {
+                    let finalDestinationMarker = GMSMarker(position: coordinates.coordinate)
+                    
+                    if let locationMarker = mapLocationMarker() {
+                        finalDestinationMarker.icon = locationMarker
+                    }
+                    finalDestinationMarkers.append(finalDestinationMarker)
+                    finalDestinationMarker.map = mapView
+                }
+            }
+        }
+
+        // Map each route segment and draw final location marker for the last segment
+        mapRouteSegment(firstRouteSegment, to: firstWalkSegment, addMarker: finalRouteSegment.isEmpty)
+        if !finalRouteSegment.isEmpty {
+            mapRouteSegment(finalRouteSegment, to: finalWalkSegment, addMarker: true)
+        }
+
+        // Configure polylines for each walking segment
+        func configurePolyline(for path: GMSPath) {
```

Comment:
Usually we don't want to nest functions in other functions just so its easier to read. I would recommend defining it right under the function that calls it